crates/matrix-sdk-crypto/src/error.rs

```diff
@@ -105,49 +105,63 @@ pub enum MegolmError {
     Store(#[from] CryptoStoreError),
 }
 
+/// Error that occurs when decrypting an event that is malformed.
 #[derive(Error, Debug)]
 pub enum EventError {
+    /// The Olm message has a unsupported type.
     #[error("the Olm message has a unsupported type, got {0}, expected 0 or 1")]
     UnsupportedOlmType(u64),
 
+    /// The Encrypted message has been encrypted with a unsupported algorithm.
     #[error("the Encrypted message has been encrypted with a unsupported algorithm.")]
     UnsupportedAlgorithm,
 
+    /// The provided JSON value isn't an object.
     #[error("the provided JSON value isn't an object")]
     NotAnObject,
 
+    /// The Encrypted message doesn't contain a ciphertext for our device.
     #[error("the Encrypted message doesn't contain a ciphertext for our device")]
     MissingCiphertext,
 
+    /// The Encrypted message is missing the signing key of the sender.
     #[error("the Encrypted message is missing the signing key of the sender")]
     MissingSigningKey,
 
+    /// The Encrypted message is missing the sender key.
     #[error("the Encrypted message is missing the sender key")]
     MissingSenderKey,
 
+    /// The Encrypted message is missing a field.
     #[error("the Encrypted message is missing the field {0}")]
     MissingField(String),
```

Comment:
This one seems to be dead code as well but I can't remember when the death occurred.